devito/ir/iet/analysis.py

```diff
@@ -70,6 +71,12 @@ def mark_parallel(analysis):
         for depth, i in enumerate(tree):
             if i in properties:
                 continue
+
+            if i.uindices:
```

Comment:
I think this does something else, mainly droping omp if the loop looks bad. This is a good thing, but the change in #553 only forces the compiler to only write out the only required indices in the loop (and drop all the %1 of unused indices).
So I guess this would be on top of my changes but still requires #553